baseplate/integration/pyramid/__init__.py

```diff
@@ -218,7 +218,7 @@ def _start_server_span(self, request, name, trace_info=None):
 
     def _get_trace_info(self, headers):
         extracted_values = TraceInfo.extract_upstream_header_values(TRACE_HEADER_NAMES, headers)
```

Comment:
I think the fix is to change `extract_upstream_header_values` to always return the `bytes` form rather than checking if it's `bytes` or `str`

Reply:
Or always unicode. Either way. Encoding issues like this are always a bit dodgy. We should be sure we understand what cases causes it to be bytes sometimes and unicode other.

Reply:
@pacejackson @spladug  or can I just change it to `int(extracted_values.get("sampled")) == 1` 
If I change some `extract_upstream_header_values ` function maybe some services are using this function and use the result to compare to str/bytes and break something?

I do not have any idea what cause it to be bytes.

Reply:
I _think_ it might just be a mistake?  Thrift returns `bytes` and when this was implemented, they may have assumed that was the case for HTTP as well, I think the core issue is that `extract_upstream_header_values` is returning inconsistent types and that is the issue to address.

Reply:
The reason this is fine in python2 is that it probably returns `str` in python 2 where `"1" == b"1"`